django/db/migrations/autodetector.py

```diff
@@ -1367,13 +1367,39 @@ def generate_added_constraints(self):
             app_label,
             model_name,
         ), alt_constraints in self.altered_constraints.items():
+            dependencies = []
+
             for constraint in alt_constraints["added_constraints"]:
+                model_state = self.to_state.models[app_label, model_name]
+
+                fields = list(model_state.fields.values())
+
+                fields += [
+                    field.remote_field
+                    for relations in self.to_state.relations[
+                        app_label, model_name
+                    ].values()
+                    for field in relations.values()
+                ]
```

Comment:
I don't think that this part is necessary and the tests seem to agree since they pass without these lines applied.

It makes sense to have the constraint depend on all relational field defined for the model the constraint is being added to avoid the complexity of digging in `constraint.fields` and `.condition` but there should be no need to depend on fields that are referred by other models or at least the reported problem and the provided test case doesn't relate to this use case.